frontend/src/app/main/data/workspace.cljs

```diff
@@ -547,7 +549,7 @@
             name               (cfh/generate-unique-name base-name unames :suffix-fn suffix-fn)
             objects            (update-vals (:objects page) #(dissoc % :use-for-thumbnail))
 
-            main-instances-ids (set (keep #(when (ctk/main-instance? (val %)) (key %)) objects))
+            main-instances-ids (set (keep #(when (ctc/main-instance? (val %)) (key %)) objects))
```

Comment:
This is a minor one, and consider as you want. The most efficient way to write this code is:

```clojure
(reduce-kv (fn [result id shape]
             (if (ctc/main-instance? shape)
               (conj result id)
               result))
           #{}
           objects)        
```